File: src/core/interfaces/IEnemy.ts
// === core/interfaces/IEnemy.ts ===
import Phaser from 'phaser';
import { EnemyConfig } from '../../data/types';

export interface IEnemy {
  readonly id: string;
  readonly config: EnemyConfig;
  currentHP: number;
  readonly sprite: Phaser.GameObjects.Sprite;

  takeDamage(damage: number): void;
  setTarget(target: Phaser.Math.Vector2): void;
  activate(x: number, y: number, statMultiplier: number): void;
  deactivate(): void;
  update(delta: number): void;
}

File: Assets/_Project/Scripts/Core/Interfaces/IEnemy.cs
using UnityEngine;

namespace VampireSurvivors.Core
{
    /// <summary>
    /// 敵人介面 — 所有敵人類型皆須實作此介面，
    /// 以便新增敵人類型時無需修改現有程式碼。
    /// </summary>
    public interface IEnemy
    {
        /// <summary>敵人的 ScriptableObject 設定檔。</summary>
        EnemyConfigSO Config { get; }

        /// <summary>敵人當前生命值。</summary>
        float CurrentHP { get; }

        /// <summary>
        /// 對敵人造成傷害。
        /// </summary>
        /// <param name="damage">傷害數值。</param>
        void TakeDamage(float damage);

        /// <summary>
        /// 設定敵人的追蹤目標（通常為玩家角色）。
        /// </summary>
        /// <param name="target">追蹤目標的 Transform。</param>
        void SetTarget(Transform target);

        /// <summary>
        /// 將敵人回收至物件池。
        /// </summary>
        void ReturnToPool();
    }
}
